### PR TITLE
Added Segmented sieve in C, C++ and, README.md

### LINKED ISSUE #1115: Segmented Sieve method in C, C++

### PR DESCRIPTION
**Pull request by Vaibhavraj10**
Fixes #1115 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

- Added Segmented Sieve method in C and C++
- Added README.md

#### Languages Used:
- C
- C++

#### Files Added:
- Segmented_Sieve.c
- Segmented_Sieve.cpp
- README.md

### CODE REVIEW

File: Segmented_Sieve/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+## Algorithm
+To find all the prime numbers in a given interval by Segmented Sieve method:
+
+- find all the primes up to sqrt(b), call them primes[]
+- create a boolean array is_prime[] with length = b-a+1 and fill it with true
+- for each p in primes set is_prime[i*p - a] = false starting at i=ceil(a/p)
+- for each is_prime[i]=true print i+a
+
+## Implementation
+
```
Comment (somya-kapoor) [inline Segmented_Sieve/README.md]: Please add a pseudo code

**Comment by somya-kapoor** [comment]
@Vaibhavraj10 Good work.
 The code looks good to me, but please add pseudo code.

**Comment by Vaibhavraj10** [comment]
> @Vaibhavraj10 Good work.
> The code looks good to me, but please add pseudo code in readme file.

Hey @somya-kapoor ! I have updated the above issue as you mentioned. Please do take a look at the updated README.md.

File: Segmented_Sieve/README.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+
+## Algorithm
+To find all the prime numbers in a given interval by Segmented Sieve method:
+- Use Simple Sieve to find all primes upto square root of ‘n’ and store these primes in an array “prime[]”. Store the found primes in an array ‘prime[]’.
+- We need all primes in range [0..n-1]. We divide this range in different segments such that size of every segment is at-most √n
+- Do following for every segment [low..high]
+    - Create an array mark[high-low+1]. Here we need only O(x) space where x is number of elements in given range.
+    - Iterate through all primes found in step 1. For every prime, mark its multiples in given range [low..high].
```
Comment (MastersAbh) [inline Segmented_Sieve/README.md]: Hi, please follow indentation in line 12 & 13 and then it is good to merge.

**Comment by somya-kapoor** [comment]
@vaibhav4595  Please change the file name.(Segmented_Sieve/Segmented_sieve.c to Segmented_Sieve/Segmented_Sieve.c

File: Segmented_Sieve/README.md (hunk 2 above)
Comment (Vaibhavraj10) [reply Segmented_Sieve/README.md]: Hi, @MastersAbh  ! Actually I have followed the indentation but lines 12 and 13 are the sub bullet points of lines 11. So, these two points are under that line 11.

**Comment by Vaibhavraj10** [comment]
> @Vaibhavraj10  Please change the file name.(Segmented_Sieve/Segmented_sieve.c to Segmented_Sieve/Segmented_Sieve.c)

@somya-kapoor Done!

File: Segmented_Sieve/README.md (hunk 2 above)
Comment (MastersAbh) [reply Segmented_Sieve/README.md]: Oh Sure !

**Comment by Vaibhavraj10** [comment]
@jainaman224 Please take a look at this.

**Pull request merged**